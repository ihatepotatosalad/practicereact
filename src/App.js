
import Gallery from './components/Gallery';
import './App.css';
import { React, useContext, useEffect, useState } from 'react';

function App() {
  let [count, setCount] = useState(0)
  function add() {
    setCount(count + 1)
  }
  function sub() {
    setCount(count - 1)
  }
  return (
    <div className="App">
      <button onClick={sub}>-</button>
      <h2 title="count">{count}</h2>
      <button onClick={add}>+</button>
      <Gallery />
    </div>
  );
}

export default App;
