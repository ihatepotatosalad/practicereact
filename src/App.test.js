import { toBeVisible, toHaveAttribute } from '@testing-library/jest-dom/dist/matchers';
import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer'

import userEvent from '@testing-library/user-event'

test('renders a snapshot', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('test buttons to add and subtract values', () => {
  render(<App />);
  const plusbutton = screen.getByText('+');
  const minusbutton = screen.getByText('-');
  const countDisplay = screen.getByTitle('count')
  expect(plusbutton).toBeInTheDocument()
  expect(minusbutton).toBeInTheDocument()
  expect(countDisplay).toBeInTheDocument()
  expect(countDisplay).toHaveTextContent('0')
  fireEvent.click(plusbutton)
  expect(countDisplay).toHaveTextContent('1')
  fireEvent.click(minusbutton)
  expect(countDisplay).toHaveTextContent('0')

});