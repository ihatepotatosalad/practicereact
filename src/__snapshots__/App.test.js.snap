// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="App"
>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <h2
    title="count"
  >
    0
  </h2>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <div />
</div>
`;
